machine/events/toeslag/aggregate.py

```diff
@@ -0,0 +1,322 @@
+"""
+Toeslag Aggregate
+
+Representeert de staat van een toeslag voor een burger voor een specifiek berekeningsjaar.
+Bijvoorbeeld: de zorgtoeslag van burger 123456789 voor het jaar 2025.
+
+Het object is generiek (volgt AWIR-proces) maar heeft een specifiek type dat aangeeft
+welke toeslag het is (ZORGTOESLAG, HUURTOESLAG, etc.).
+
+Deze aggregate volgt het AWIR-proces:
+1. Aanvraag ingediend
+2. Aanspraak berekend
+3. Voorschotbeschikking vastgesteld
+4. Maandelijkse herberekeningen en betalingen
+5. Definitieve beschikking
+6. Vereffening (nabetaling of terugvordering)
+"""
+
+from datetime import date, datetime
+from enum import Enum
+from typing import Any
+
+from eventsourcing.domain import Aggregate, event
+
+
+class ToeslagType(str, Enum):
+    """Type toeslag - bepaalt welke regeling wordt gebruikt"""
+
+    ZORGTOESLAG = "ZORGTOESLAG"
+    HUURTOESLAG = "HUURTOESLAG"
+    KINDGEBONDEN_BUDGET = "KINDGEBONDEN_BUDGET"
+    KINDEROPVANGTOESLAG = "KINDEROPVANGTOESLAG"
+
+
+# Mapping van type naar regeling (wet)
+TOESLAG_TYPE_REGELING = {
+    ToeslagType.ZORGTOESLAG: "zorgtoeslagwet",
+    ToeslagType.HUURTOESLAG: "wet_op_de_huurtoeslag",
+    ToeslagType.KINDGEBONDEN_BUDGET: "wet_op_het_kindgebonden_budget",
+    ToeslagType.KINDEROPVANGTOESLAG: "wet_kinderopvang",
+}
+
+
+class ToeslagStatus(str, Enum):
+    """Status van de toeslag in het AWIR-proces"""
+
+    AANVRAAG = "AANVRAAG"
+    BEREKEND = "BEREKEND"
+    VOORSCHOT = "VOORSCHOT"
+    LOPEND = "LOPEND"  # Maandelijkse cyclus actief
+    DEFINITIEF = "DEFINITIEF"
+    VEREFFEND = "VEREFFEND"
+    AFGEWEZEN = "AFGEWEZEN"
+    BEEINDIGD = "BEEINDIGD"
+
+
+class Toeslag(Aggregate):
```

Comment:
Het is wel jammer om domeinspecifieke classes te introduceren, zou moeten volgen uit een yaml file met interpretatie wet / regels